parlai/tasks/convai2/agents.py

```diff
@@ -105,93 +105,11 @@ def normalize_replies(self, x):
-                # x<tab>y<tab>reward<tab>label_candidates
-                # where y, reward, and label_candidates are optional
-                split = line[space_idx + 1 :].split('\t')
-
-                # remove empty items and strip each one
-                for i in range(len(split)):
-                    word = split[i].strip()
-                    if len(word) == 0:
-                        split[i] = ''
-                    else:
-                        split[i] = word
-                # Empty reward string same as None
-                if len(split) > 2 and split[2] == '':
-                    split[2] = None
-
-                # now check if we're at a new episode
-                if last_conv_id is None or conv_id <= last_conv_id:
-                    x = x.strip()
-                    if x:
-                        x = self.normalize_replies(x)
-                        import pdb
-
-                        pdb.set_trace()
-                        yield [x, None, reward], start
-                    start = True
-                    reward = 0
-                    # start a new episode
-                    x = split[0]
-                else:
-                    if x:
-                        # otherwise add current x to what we have so far
-                        x = '{x}\n{next_x}'.format(x=x, next_x=split[0])
-                    else:
-                        x = split[0]
-                last_conv_id = conv_id
-                if len(split) > 2 and split[2]:
-                    reward += float(split[2])
-
-                if len(split) > 1 and split[1]:
-                    # only generate an example if we have a y
-                    split[0] = x
-                    # split labels
-                    split[1] = split[1].split('|')
-                    if len(split) > 3:
-                        # split label_candidates
-                        split[3] = split[3].split('|')
-                    if len(split) > 2:
-                        split[2] = reward
-                    else:
-                        split.append(reward)
-                    # normalize
-                    split[0] = self.normalize_replies(split[0])
-                    for i, _c in enumerate(split[1]):
-                        split[1][i] = self.normalize_replies(split[1][i])
-                    for i, _c in enumerate(split[3]):
-                        split[3][i] = self.normalize_replies(split[3][i])
-                    if start:
-                        yield split, True
-                        start = False
-                    else:
-                        yield split, False
-                    # reset x in case there is unlabeled data still left
-                    x = ''
-                    reward = 0
-            if x:
-                x = self.normalize_replies(x)
-                yield [x, None, reward], start
+        for (text, labels, reward, candidates), new_episode in super().setup_data(path):
```

Comment:
Hmm - is this this duplicated logic? Great that we can remove so much formatting code...